### PR TITLE
Allows feeding viscerite to synthflesh pustules

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
## About the PR 
Feeding pustules viscerite now gives the pustule 20 u of viscerite viscera per material. This chem heals 2.5 brute and burn, but deals +1.5 tox, purges chems at the same rate as calomel, and occasionally gives you weakened. The intent here is for it to serve as essientally a reverse pententic acid.

The pustule proceeds to digest the viscera into toxic martian flesh and synthflesh. In the process of digestion, the pustule occasionally emits sulfuric acid, with small/medium pustules doing so in relatively higher quantities with increased frequency. Small/medium pustules produce 7 units of synthflesh and martian flesh, alongside (on average) 4u of sulfuric acid per viscerite, large pustules produce 14 units synthflesh and 2.5 units martian flesh, alongside (on average) 1 units sulfuric acid.

## Why's this needed? 
Gives some neat usage to viscerite, interdepartmental cooperation and all that alongside making some cool new chems available. Also just seems like a neat and logical interaction,

## Changelog 

```changelog
(u)Ikea
(*)Feeding viscerite to synthflesh pustules generates viscerite viscera. The pustule proceeds to digest this chemical into synthflesh and martian flesh, all while offgassing sulfuric acid. Small/medium pustules produce 7 units of synthflesh and 7 units of martian flesh per viscerite, large pustules produce 14 units of synthflesh and 2.5 units of martian flesh per viscerite.
(+) A new chem has been added in the form of viscerite viscera. Currently only obtainable from synthflesh pustules, this chem heals 2.5 brute and burn, all while dealing 1.5 tox, purging other chems, and dealing weakened.
```

### CODE REVIEW

File: code/modules/chemistry/Chemistry-Tools.dm [hunk 1]
```diff
@@ -1017,6 +1018,18 @@ proc/ui_describe_reagents(atom/A)
 				reagents.add_reagent("synthflesh", 40 * organ_efficiency)
 				become_unangry()
 
+		if(W.material == getMaterial("viscerite"))
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Chemistry-Tools.dm:1021]:
```suggestion
		if(W.material?.isSameMaterial(getMaterial("viscerite")))
```
== doesn't account for weird mutable material cases

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Neat idea, but I think it should come with some kind of downside to offset how easy it is to get a large quantity of viscerite. Maybe make it add some kind of toxic contaminant or emit weird eldritch smoke. Also probably reduce the efficiency a bit.

File: code/modules/chemistry/Chemistry-Tools.dm [hunk 2]
```diff
@@ -1017,6 +1029,37 @@ proc/ui_describe_reagents(atom/A)
...
+					else
+						amt_to_remove = visc.amount
+					reagents.add_reagent("viscerite_viscera", 20 * amt_to_remove * visc.material_amt)
+					visc.amount -= amt_to_remove
+					if (visc.amount <= 0)
+						qdel(visc)
+					else
+						if(visc.inventory_counter)
+							visc.inventory_counter.update_number(visc.amount)
+							visc.UpdateStackAppearance()
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Chemistry-Tools.dm:1057]: A lot of this mess could likely be replaced with `change_stack_amount`

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Hmmm, interesting idea but at that point why not just have the pustule keep the new chem instead of converting into synthflesh?

**Comment by ZeWaka** [comment]
x

File: code/modules/chemistry/Chemistry-Tools.dm [hunk 3]
```diff
@@ -994,8 +994,19 @@ proc/ui_describe_reagents(atom/A)
...
+			if("viscerite_viscera")
+				reagent_to_add = null
+		if(reagent_to_add)
+			src.reagents.add_reagent(reagent_to_add, volume * chemical_efficiency)
+
+		if(reagent_id == "viscerite_viscera") // Special processing is needed for this chem because its my birthday and I get to have feature bloat
+			src.reagents.add_reagent("martian_flesh", ((volume / chemical_efficiency) * 0.35)) // Large pustules produce half the amount of martian flesh. Apologies for the magic number here, it's avoiding digestion producing net amounts of chems which leads to bugs.
+			src.reagents.add_reagent("synthflesh", ((volume * chemical_efficiency)  * 0.35)) // Large pustules also produce twice the amount of synthflesh
+			if(prob(20 / chemical_efficiency)) // 20% chance of offgassing for medium/small, 10% for large pustules
+				var/datum/reagents/smokeContents = new/datum/reagents/
```
Comment (ZeWaka) [inline code/modules/chemistry/Chemistry-Tools.dm:1006]:
```suggestion
				var/datum/reagents/smokeContents = new/datum/reagents()
```
gross

File: code/modules/chemistry/Reagents-Misc.dm [hunk 4]
```diff
@@ -2094,6 +2094,39 @@ datum
 					if(method == TOUCH)
 						boutput(M, SPAN_ALERT("Well, that was gross."))
 
+		viscerite_viscera
+			name = "viscerite viscera"
+			id = "viscerite_viscera"
+			description = "Looking at this makes you feel queasy"
```
Comment (ZeWaka) [inline code/modules/chemistry/Reagents-Misc.dm:2100]:
```suggestion
			description = "Looking at this makes you feel queasy... ugh."
```

**Review by ZeWaka — COMMENTED** [review]
unsure on the feature but code looks alright enough

**Comment by ZeWaka** [comment]
@TobleroneSwordfish

**Pull request merged**